### DISCUSSION TITLE
Laravel Pulse auto clean-up

### DISCUSSION THREAD

**Discussion by didac-adria** [laravel/framework#50100] [Q&A] [answered]
Does anybody know if Pulse auto-cleans old records? Reading the documentation, makes me think it does. But reality is that in our case, DB records are piling up. Should we create a scheduled task to delete them? pulse_aggregates table though don't contain a timestamp. However, we think the "bucket" table is a timestamp which could be used to know which records to delete.

I think it would be great if Pulse deletes old records and is well documented in the documentation.

**Comment by ahinkle** [comment, marked as answer]
> Pulse will automatically trim its stored entries once they are outside of the dashboard window. Trimming occurs when ingesting data using a lottery system which may be customized in the Pulse configuration file.

https://laravel.com/docs/10.x/pulse#trimming

**Reply by didac-adria** [reply]
Hey @ahinkle! Yes, I've seen that in the documentation. But the reality is that is not happening in our database, although it's in our config file. Any idea why?

**Reply by ahinkle** [reply]
What's your config file look like?

**Reply by didac-adria** [reply]
We are using the default configuration. We use Forge and a deamon to run the pulse:check command. We use the DB to save records. I've opened an issue here, looks like there could be a bug which is yet not confirmed:

https://github.com/laravel/pulse/issues/317#issuecomment-1948271914

**Reply by didac-adria** [reply]
But just for the record:

```
'storage' => [
        'driver' => env('PULSE_STORAGE_DRIVER', 'database'),

        'database' => [
            'connection' => env('PULSE_DB_CONNECTION', null),
            'chunk' => 1000,
        ],
    ],

'ingest' => [
        'driver' => env('PULSE_INGEST_DRIVER', 'storage'),

        'buffer' => env('PULSE_INGEST_BUFFER', 5_000),

        'trim' => [
            'lottery' => [1, 1000],
            'keep' => '7 days',
        ],

        'redis' => [
            'connection' => env('PULSE_REDIS_CONNECTION'),
            'chunk' => 1000,
        ],
    ],
```

**Reply by cAstraea** [reply]
Can we update the docs on how to use this lottery config please ? A bit hard to understand. Tried messing around but still ending up with 1GB of data in the pulse tables ... for now have a job that truncates everything every half a day..
